### DISCUSSION TITLE
Which stanza is correct for this situation?

### DISCUSSION THREAD

**Discussion by borrrden** [Homebrew/discussions#2827] [Casks] [answered]
I develop a library which is unfortunately not 100% open source, but I'd like to put it onto Homebrew if possible so I am looking into the Cask route.  However, I don't see any stanza that does quite what I want it to do, besides the seemingly discouraged `artifact`.  

My deliverable is two folders:  `include` containing header files and `lib` containing CMake config and shared libraries.  The closest stanza I see is the `binary` stanza which installs things into `<prefix>/bin` but I want to simply have `<prefix>/include` and `<prefix>/lib`.  Is my only choice to write a `pkg` file or to use the `artifact` stanza?

**Comment by miccal** [comment, marked as answer]
The `artifact` stanza is not discouraged, and that is the one you want I think -- please go ahead and create a PR so we can review it.

**Reply by borrrden** [reply]
Sorry if this question has an obvious answer, but is there a way to know the `brew --prefix` to use inside of the target path of the artifact?  Basically the behavior I'm looking for is that of a normal formula which builds and then symlinks things in.  The best I can do now is have everything copied arbitrarily into `/usr/local` minus the library symlinks (i.e. `libX.3.dylib -> libX.3.0.0.dylib`)

**Reply by gromgit** [reply]
> is there a way to know the `brew --prefix` to use inside of the target path of the artifact?

You want `HOMEBREW_PREFIX`. Search for that term in `$(brew --prefix homebrew/cask)/Casks/*` to see how it's used.

**Reply by carlocab** [reply]
*`$(brew --repository homebrew/cask)/Casks/*`

**Reply by gromgit** [reply]
Whoops. I swear I knew that. :laughing:

**Reply by borrrden** [reply]
Thanks, that combined with postflight is all I needed to get things to look equivalent to the formula route.  I believe I'll be ready to send a PR once we put out a GA package of the product.
